Library/Homebrew/dependency_collector.rb

```diff
@@ -179,6 +179,8 @@ def resource_dep(spec, tags)
     if strategy <= HomebrewCurlDownloadStrategy
       @deps << curl_dep_if_needed(tags)
       parse_url_spec(spec.url, tags)
+    elsif strategy <= NoUnzipCurlDownloadStrategy
+      # allow no unzip to pass through
```

Comment:
```suggestion
      # ensure NoUnzip never adds any dependencies
```

Reply:
This should be above the previous check.